source/utilities.cc

```diff
@@ -2978,5 +3054,18 @@ namespace aspect
                                                const unsigned int n_rows,
                                                const unsigned int n_columns,
                                                const std::string &property_name);
+
+    template std::vector<std::size_t>
+    compute_sorting_permutation(const std::vector<double> &vector);
```

Comment:
I am thinking these two functions might be better to be inline functions with the implementation in the h file. Otherwise you can only use it with this specific type. Maybe somebody else has an opinion on the usefulness of this?